core/src/main/java/jenkins/triggers/ReverseBuildTrigger.java

```diff
@@ -222,14 +225,14 @@ public FormValidation doCheckUpstreamProjects(@AncestorInPath Job project, @Quer
     }
 
     @Extension public static final class RunListenerImpl extends RunListener<Run> {
-        @Override public void onCompleted(Run r, TaskListener listener) {
+        @Override public void onCompleted(@Nonnull Run r, @Nonnull TaskListener listener) {
```

Comment:
violates the contract on the superclass (where ironically Run can be null :-)  )